java-checks/src/test/files/checks/LazyArgEvaluationCheck.java

```diff
@@ -0,0 +1,142 @@
+    slf4j.error("Unable to open file " + csvPath, new RuntimeException());  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+
+    slf4j.error("Unable to open file " + csvPath, new RuntimeException());  // Noncompliant
+    slf4j.error("Unable to open file " + csvPath, new RuntimeException(), 1);  // Noncompliant
+    slf4j.error("Unable to open file " + csvPath, new RuntimeException(), 1, 2);  // Noncompliant
+    slf4j.error("Unable to open file " + csvPath, new RuntimeException(), 1, "vargs", "vargs", "vargs FTW!");  // Noncompliant
+
+    Marker confidentialMarker = MarkerFactory.getMarker("CONFIDENTIAL");
+    slf4j.error(confidentialMarker, "Unable to open file " + csvPath, new RuntimeException());  // Noncompliant
+    slf4j.error(confidentialMarker, "Unable to open file " + csvPath, new RuntimeException(), 1);  // Noncompliant
+    slf4j.error(confidentialMarker, "Unable to open file " + csvPath, new RuntimeException(), 1, 2);  // Noncompliant
+    slf4j.error(confidentialMarker, "Unable to open file " + csvPath, new RuntimeException(), 1, "vargs", "vargs", "vargs FTW!");  // Noncompliant
+
+    if (slf4j.isTraceEnabled()) {
+      slf4j.trace("Unable to open file " + csvPath, new RuntimeException());  // Compliant - inside if test
+    }
+    if (slf4j.isDebugEnabled()) {
+      slf4j.debug("Unable to open file " + csvPath, new RuntimeException());  // Compliant - inside if test
+    }
+    if (slf4j.isInfoEnabled()) {
+      slf4j.info("Unable to open file " + csvPath, new RuntimeException());  // Compliant - inside if test
+    }
+    if (slf4j.isWarnEnabled()) {
+      slf4j.warn("Unable to open file " + csvPath, new RuntimeException());  // Compliant - inside if test
+    }
+    if (slf4j.isErrorEnabled()) {
+      slf4j.error("Unable to open file " + csvPath, new RuntimeException());  // Compliant - inside if test
+    }
+  }
+  
+  void jul() {
+    logger.finest("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.finer("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.fine("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.config("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.info("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.warning("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+    logger.severe("Unable to open file " + csvPath);  // Noncompliant {{Use the built-in formatting to construct this argument.}}
+
+    if (logger.isFinestEnabled()) {
```

Comment:
There is something fishy with your implementation, and in particular your if-condition visitor (especially since the name is actually not matching).
The following code does NOT raise an issue, where I would expect one (you actually don't test this case...):
```
  void foo(boolean b) {
    if (b) {
      logger.finest("Unable to open file " + csvPath);  // Noncompliant
    }
  }
```